apps/settings/js/modules/storage/device_storages.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * DeviceStorages:
+ *   - DeviceStorages is an Observable that wraps the platform
+ *     DeviceStorage object.
+ *   - DeviceStorages is a singleton that you can easily use it to fetch some
+ *     shared data across different panels.
+ *   - It has some observable properties: todo: state, enabled, address, name,
+ *   - It has one observable array: _volumesList.
+ * DeviceStorages only update state and does not involve in any UI logic.
+ *
+ * @module DeviceStorages
+ */
+define(function(require) {
+  'use strict';
+
+  var Volume = require('modules/storage/volume');
+  var ObservableArray = require('modules/mvvm/observable_array');
+
+  const MEDIA_TYPE = ['music', 'pictures', 'videos', 'sdcard'];
+  const FIRST_VOLUME_NAME = 'sdcard';
+
+  var _debug = false;
+  var Debug = function() {};
+  if (_debug) {
+    Debug = function ds_debug(msg) {
+      console.log('--> [DeviceStorages]: ' + msg);
+    };
+  }
+
+  var DeviceStorages = {
```

Comment:
nit: how about `DeviceStorageManager`? `DeviceStorages` seems imply that the module is actually an array but it is not.